### DISCUSSION TITLE
How to dynamically import  metadata  wit next/mdx

### DISCUSSION THREAD

**Discussion by oeyoews** [vercel/next.js#53553] [App Router]
![image](https://github.com/vercel/next.js/assets/72405338/55943f77-55aa-4767-9598-d82b19c38f93)

```tsx
export const metadata = {
  title: "Hello",
  description: "This is a description",
}
```

I try to import the metadata object, but dynamic seems to support functions with return values, not support import constants, can I only try the following way of writing, this doesn't look like a good idea(I am trying how to dynamically read the metadata of the page.mdx page with @next/mdx dynamicly)

```jsx
export const metadata = {
  title: () => "Hello",
  description: () => "This is a description",
}
```

```jsx
dynamic(() => import('./hello').then((mod) => mod.metadata.title))
```

**Comment by jeandv** [comment]
You can then access the metadata in your Next.js page by importing the .mdx file and accessing the meta property. Here’s an example:

```
import Content, { meta } from './my-mdx-page.mdx'

export default function Page() {
  return (
    <>
      <h1>{meta.author}</h1>
      <Content />
    </>
  )
}
```
next/dynamic is used for dynamically importing React components, not for importing constants or objects] If you want to dynamically import the metadata object, you can use the regular dynamic import() syntax. Here’s an example:

```
import { useState, useEffect } from 'react'

export default function Page() {
  const [metadata, setMetadata] = useState({})

  useEffect(() => {
    async function loadMetadata() {
      const data = await import('./my-mdx-page.mdx')
      setMetadata(data.meta)
    }
    loadMetadata()
  }, [])

  return <h1>{metadata.author}</h1>
}
```

I hope I helped you, I was guided by the next.js documentation and AI help.
